### ISSUE TITLE
Consider disabling unused repositories in minimal templates

### ISSUE THREAD

**Issue by donob4n** [QubesOS/qubes-issues#6186]
**The problem you're addressing (if any)**

From a security perspective the most critical attack vector for a template is their enabled repositories so having some enabled without a real need is just an avoidable risk.

**Describe the solution you'd like**

Disable fedora-cisco-openh264.repo and probably fedora-modular too from fedora-XX-minimal

Maybe adjust the documentation if some of them should be enabled again for some purpose.

**Comment by andrewdavidwong** [comment]
- If a repo is enabled but unused, then I'm guessing the (only) threat is from potentially malicious repo metadata being downloaded and parsed by dnf or apt.

 - When I install packages in a minimal template, I carefully select the minimum number of packages I need, and I write them down in a list in case I need to re-create the template in the future. However, I have no idea which repo my package was pulled from, and I have no idea which repos I need to keep enabled in order to receive updates for the software installed in a given template. I wouldn't want to disable a repo and not receive important security updates as a result. I checked `dnf history`, and there's no indication of which repo(s) were used. Is such information not available? If not, how are we (users) supposed to manage this?

**Comment by donob4n** [comment]
>   If a repo is enabled but unused, then I'm guessing the (only) threat is from potentially malicious repo metadata being downloaded and parsed by dnf or apt.

But if some repository is compromised (e.g. fedora-cisco-openh264) could it force you to upgrade a package that you installed from another repository?
 
> I checked `dnf history`, and there's no indication of which repo(s) were used. Is such information not available? If not, how are we (users) supposed to manage this?

With 'dnf list' you get a list of installed packages and the third column shows the source of the package.

**Comment by andrewdavidwong** [comment]
> But if some repository is compromised (e.g. fedora-cisco-openh264) could it force you to upgrade a package that you installed from another repository?

Not sure. @marmarek, what do you think?

> With 'dnf list' you get a list of installed packages and the third column shows the source of the package.

Ah, that's a handy tip. Thanks!

**Comment by marmarek** [comment]
> But if some repository is compromised (e.g. fedora-cisco-openh264) could it force you to upgrade a package that you installed from another repository?

Yes, dnf does not consider source repository when looking for update - any repository can ship update for example to bash package.
So, yes, generally I consider this a good idea, although it is against https://www.qubes-os.org/faq/#what-is-qubes-attitude-toward-changing-guest-distros. Same as some other exclusion in -minimal template.
@donob4n can you prepare an update to minimal template documentation that those two repositories are disabled by default compared to the vanilla Fedora? Otherwise this will definitely lead to user confusion ("why package X fails to install in minimal template even though it works in the full one?").

**Comment by donob4n** [comment]
Ok!

**Comment by DemiMarie** [comment]
For this to be a significant security benefit, minimal templates will need the ability to run Salt, which they currently do not by default.

**Comment by andrewdavidwong** [comment]
> For this to be a significant security benefit, minimal templates will need the ability to run Salt, which they currently do not by default.

1. Why does that follow? The security benefit of disabling unused repos applies whenever a template is updated, and all templates have to be updated, whether they run Salt or not.

2. Minimal templates can run Salt simply by installing the required package. I'm not sure why it needs to be by default. That's the nature of minimal templates: They can hardly do anything by default. That's intentional. They're a minimal base on which to add your own desired packages so that they can be useful.

**Comment by DemiMarie** [comment]
> > For this to be a significant security benefit, minimal templates will need the ability to run Salt, which they currently do not by default.
> 
>     1. Why does that follow? The security benefit of disabling unused repos applies whenever a template is updated, and all templates have to be updated, whether they run Salt or not.
Salt is the recommended and default way to update a template.  Therefore, to gain the security benefit, a minimal VM needs to be the template for its own Management DVM, which means it needs the Salt package installed.

>     2. Minimal templates can run Salt simply by installing the required package. I'm not sure why it needs to be by default. That's the nature of minimal templates: They can hardly do anything by default. That's intentional. They're a minimal base on which to add your own desired packages so that they can be useful.

I would like to see the minimal template be the default template for the default Management DispVM, which requires that it have Salt installed.

**Comment by andrewdavidwong** [comment]
> Salt is the recommended and default way to update a template. Therefore, to gain the security benefit, a minimal VM needs to be the template for its own Management DVM, which means it needs the Salt package installed.

If I understand correctly, the theoretical attack is a malicious package coming from a repo that the user does not otherwise use or want to use. Disabling that repo closes off that avenue of attack. If this is correct, then it is immaterial whether the update was triggered by Salt or other means. The security benefit of disabling the unused repos is the same either way.

Rather, I understand your point to be that we should ensure that this security benefit applies when templates are updated via the default method. In other words, we should strive to set up secure defaults. I agree wholeheartedly.

> I would like to see the minimal template be the default template for the default Management DispVM, which requires that it have Salt installed.

I think that would be beneficial. (That's the setup I personally use, having installed the required package manually.) However, the  potential snag is that we can't make a minimal template the default template that most users are supposed to use for their AppVMs. In other words, it can't simply replace the default full Fedora template. So, this would probably require adding yet another template to an already-large ISO (a minimal one, but still).

**Comment by DemiMarie** [comment]
We could also create a hyperminimal “micro” template that only includes the bare minimum needed to boot and run Salt ― essentially just qrexec, salt, and their transitive dependencies.  No GUI, fonts, or other niceties.

**Comment by marmarek** [comment]
That should also reduce its resources (RAM, boot time).

**Comment by DemiMarie** [comment]
A VM that is truly only for Salt can get rid of a lot more bloat: it doesn’t need a kernel, initramfs, or even systemd or networking.  That is less a hyperminimal Fedora template, though, and more a Salt-only template.

**Comment by DemiMarie** [comment]
Such a VM would probably be more like the stubdomain than anything else ― tiny, stateless, and running completely from memory

**Comment by andrewdavidwong** [comment]
Sounds good. Branched to a new issue: #6230.

**Comment by rootnoob** [comment]
Release is still TBD on this. Issue still present.
Given this just requires a tweak to the config, when building the template, can release be set to r4.1 updates and applied to new templates?

**Comment by DemiMarie** [comment]
@marmarek thoughts?

**Comment by ben-grande** [comment]
@marmarek ping
